src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1976,6 +1978,22 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
         $transportRetryReferences = [];
         foreach ($config['transports'] as $name => $transport) {
             $serializerId = $transport['serializer'] ?? 'messenger.default_serializer';
+
+            if (isset($transport['symfony_serializer'])) {
+                $serializerDefinition = $container->findDefinition($serializerId);
+
+                if (!isset(class_implements($serializerDefinition->getClass())[FormatAndContextAwareSerializerInterface::class])) {
```

Comment:
IMO it's safer to rely on an interface to inject these parameters instead of constructor's parameters order

Reply:
I don't think this interface makes sense in the Messenger package. It's way too tied to the Symfony Serializer implementation, and it's not even used at runtime, only here for DI.

> the service could be defined in user land

If the user defines the service in userland, then they can inject the format and context themselves and won't need this configuration entry, right?

I'd remove the interface, trait and method call to inject format and context using constructor's parameters, as for the `messenger.symfony_serializer` section.

Reply:
I agree with @ogizanagi here